packages/api/src/controllers/asset.test.ts

```diff
@@ -316,5 +324,58 @@ describe("controllers/asset", () => {
         },
       });
     });
+    describe("chunked upload", () => {
+      jest.setTimeout(10000);
+      it("should start upload, stop it, resume it on tus test server", async () => {
+        const filename = "test.mp4";
+        const path = "/tmp";
+        const filePath = `${path}/${filename}`;
+        const res = await client.post("/asset/request-upload", {
+          name: "tus-test",
+        });
+        await createMockFile(filePath, 1024 * 1024 * 10);
+        const file = fs.createReadStream(filePath);
+        const { size } = fs.statSync(filePath);
+        expect(res.status).toBe(200);
+        let { tusEndpoint } = await res.json();
+        tusEndpoint = tusEndpoint.split("/api/")[1];
+        tusEndpoint = `${client.server.host}/api/${tusEndpoint}`;
+        await new Promise<void>((resolve, reject) => {
+          let aborted = false;
+          const upload = new tus.Upload(file, {
+            endpoint: tusEndpoint,
+            urlStorage: new (tus as any).FileUrlStorage("/tmp/metadata"),
+            metadata: {
+              filename,
+              filetype: "video/mp4",
+            },
+            uploadSize: size,
+            onError(error) {
+              reject(error);
+              throw error;
+            },
+            onProgress(bytesUploaded, bytesTotal) {
+              const percentage = parseFloat(
+                ((bytesUploaded / bytesTotal) * 100).toFixed(2)
+              );
+              if (!aborted && percentage > 0) {
+                async () => {
+                  await upload.abort(true);
+                  aborted = true;
+                };
```

Comment:
You're not actually calling this function, only declaring it.

Also, might want to `aborted = true` before calling abort, otherwise you may get more segments uploaded before `abort` actually gets to execute. Simplifying:
```suggestion
                aborted = true;
                await upload.abort(true);
```